master/internal/db/postgres_checkpoints.go

```diff
@@ -0,0 +1,204 @@
+package db
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	_ "github.com/golang-migrate/migrate/source/file" // Load migrations from files.
+	"github.com/google/uuid"
+	_ "github.com/jackc/pgx/v4/stdlib" // Import Postgres driver.
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// AddCheckpoint adds the checkpoint to the database and sets its ID.
+func (db *PgDB) AddCheckpoint(checkpoint *model.Checkpoint) error {
+	if !checkpoint.IsNew() {
```

Comment:
We lost some correctness constraints provided by the data model here. Should we add them back in until we're ready to part with them? (e.g. the check that a step at least exists for this and should we wait until the API is exposed? even when it is, should creating a validation without a step attached to it create a metric-less step at that many batches so that the UI still remains useful exactly as it is?)

Reply:
> Should we add them back in until we're ready to part with them?

Don't think it is necessary to keep this constraint in the DB since we only record checkpoints on specific steps. So the WebUI will work fine.

We should remove them in the push architecture. I've created a [follow-up ticket](https://determinedai.atlassian.net/browse/DET-4980) to return all the checkpoints/validations for each trial.